src/dstack/_internal/proxy/gateway/services/registry.py

```diff
@@ -321,13 +329,39 @@ async def get_nginx_service_config(
         limit_req_zones.append(
             LimitReqZoneConfig(name=zone_name, key=key, rpm=round(rate_limit.rps * 60))
         )
-        locations.append(
-            LocationConfig(
-                prefix=rate_limit.prefix,
-                limit_req=LimitReqConfig(zone=zone_name, burst=rate_limit.burst),
+        if is_sglang:
+            for path in sglang_whitelisted_paths:
+                if rate_limit.prefix == path or path.startswith(rate_limit.prefix):
```

Comment:
(nit) `str_a == str_b` is a special case of `str_b.startswith(str_a)`.


```suggestion
                if path.startswith(rate_limit.prefix):
```